jena-base/src/main/java/org/apache/jena/atlas/lib/cache/CacheSimple.java

```diff
@@ -37,120 +37,81 @@
 
 public class CacheSimple<K,V> implements Cache<K,V>
 {
-    private final V[] values ; 
-    private final K[] keys ;
-    private final int size ;
-    private int currentSize = 0 ;
-    private BiConsumer<K,V> dropHandler = null ;
+	private Map<K,V> internalCache;
+	
```

Comment:
A Map grows as more, different, keys are added whereas the existing CacheSimple is a fixed size slot replacement cache with the property that it never resizes. 

CacheSimple is a fast cache, traded against a poorer than LRU replacement/eviction policy of same-hash slot replacement.  Using a Map changes its characteristics.